micro_sam/instance_segmentation.py

```diff
@@ -269,6 +269,7 @@ def _to_mask_data(self, masks, iou_preds, crop_box, original_size, points=None):
 
         # threshold masks and calculate boxes
         data["masks"] = data["masks"] > self._predictor.model.mask_threshold
+        data["masks"] = data["masks"].type(torch.int)
```

Comment:
I have traced down the failing test to this line. It seems like the bounding boxes computed by `batched_mask_to_box` are different when a `bool` tensor is passed than when an `int` tensor is passed. This is unfortunate, but would be a bit of effort to fix since this is part of upstream code.
@GenevieveBuckley I assume that you have made this change because something fails with `mps` if a bool tensor is used. Can you please send me the exact error message you get when commenting out that line? Then I will think about the best strategy here.

Reply:
1. Wow, I did not know boolean masks returned correct results, but integer masks return something totally different. Yikes! I have changed the type of this line to `torch.bool`, thank you for pointing it out.

2. I have also gone ahead with my first [suggestion here](https://github.com/computational-cell-analytics/micro-sam/pull/178#issue-1883345111) about how to handle the problem with `batched_mask_to_box()`. I have made a new file `_vendored.py` containing a copy of `batched_mask_to_box()` from `segment_anything/util/amg.py`, which I have edited to (a) make sure the input mask is boolean, and (b) make compatible with the MPS Pytorch backend for apple silicon.

I think this should fix the problems.

Reply:
> 1. Wow, I did not know boolean masks returned correct results, but integer masks return something totally different. Yikes! I have changed the type of this line to `torch.bool`, thank you for pointing it out.

Yes, it is really unfortunate... I think this is a pretty big bug on the torch / SegmentAnything side. It could be worth it to figure out what exactly causes it and report that at some point.

> 2\. I have also gone ahead with my first [suggestion here](https://github.com/computational-cell-analytics/micro-sam/pull/178#issue-1883345111) about how to handle the problem with `batched_mask_to_box()`. I have made a new file `_vendored.py` containing a copy of `batched_mask_to_box()` from `segment_anything/util/amg.py`, which I have edited to (a) make sure the input mask is boolean, and (b) make compatible with the MPS Pytorch backend for apple silicon.
> 
> I think this should fix the problems.

:+1:

Reply:
Ah, I see you made an issue already :D 
https://github.com/facebookresearch/segment-anything/issues/552